build/CI related refactoring

jpopelka opened the pull request:
Initially, I wanted to change tests/CI so that it uses pre-built tests image, which we already have in [Quay](https://quay.io/repository/packit/packit-tests).

But I realized the image is actually not used in CI (Zuul and/or Testing Farm), only when running tests locally,
so I decided it makes no sense to do such a change, reverted it and now this PR is only about few details I changed on the way.

TomasTomecek commented on Makefile:
```diff
@@ -25,7 +21,8 @@ check:
 # example: TESTS_TARGET=tests/unit/test_api.py make check_in_container
-check_in_container: tests_image
+check_in_container:
+	$(CONTAINER_ENGINE) pull $(TESTS_IMAGE)
```
both podman and docker should support `--pull` for `run`
```
   --pull=always|missing|never
       Pull image before running. The default is missing.                                                                                                                                     

              • missing: attempt to pull the latest image from the registries listed in registries.conf if a local image does not exist.Raise an error if the image is not in any  listed
                registry and is not present locally.
                                               
              • always:  Pull  the image from the first registry it is found in as listed in registries.conf. Raise an error if not found in the registries, even if the image is present
                locally.
                                                                                                                                                                                              
              • never: do not pull the image from the registry, use only the local version. Raise an error if the image is not present locally.    
```

TomasTomecek reviewed (approved):
nice, thank you for making this smoother!

jpopelka commented:
I changed my mind and changed the last commit so that we don't use the pre-built image after all and still build our own. (see updated description above)
